src/Nethermind/Nethermind.Facade/MultiCallTxTracer.cs

```diff
@@ -0,0 +1,242 @@
+            {
+                Data = entry.Data,
+                Address = entry.LoggersAddress,
+                Topics = entry.Topics,
+                LogIndex = (ulong)i
+            }).ToArray()
+        };
+    }
+
+    public void MarkAsFailed(Address recipient, long gasSpent, byte[] output, string error, Keccak? stateRoot = null)
+    {
+        TraceResult = new MultiCallCallResult()
+        {
+
+            GasUsed = (ulong)gasSpent,
+            Error = new Facade.Proxy.Models.MultiCall.Error
+            {
+                Code = StatusCode.Failure,
+                Message = error
+            },
+            Return = output,
+            Status = StatusCode.Failure.ToString(),
+            Logs = _extendedLogs.Select((entry, i) => new Log
+            {
+                Data = entry.Data,
+                Address = entry.LoggersAddress,
+                Topics = entry.Topics,
+                LogIndex = (ulong)i
+            }).ToArray()
+        };
+    }
+
+    public void StartOperation(int depth, long gas, Instruction opcode, int pc, bool isPostMerge = false)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void ReportOperationError(EvmExceptionType error)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void ReportOperationRemainingGas(long gas)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void SetOperationStack(List<string> stackTrace)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void ReportStackPush(in ReadOnlySpan<byte> stackItem)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void SetOperationMemory(List<string> memoryTrace)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void SetOperationMemorySize(ulong newSize)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void ReportMemoryChange(long offset, in ReadOnlySpan<byte> data)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void SetOperationStorage(Address address, UInt256 storageIndex, ReadOnlySpan<byte> newValue,
+        ReadOnlySpan<byte> currentValue)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void LoadOperationStorage(Address address, UInt256 storageIndex, ReadOnlySpan<byte> value)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void ReportSelfDestruct(Address address, UInt256 balance, Address refundAddress)
+    {
+    }
+
+    public void ReportAction(long gas, UInt256 value, Address from, Address to, ReadOnlyMemory<byte> input,
+        ExecutionType callType,
+        bool isPrecompileCall = false)
+    {
+        byte[]? data = AbiEncoder.Instance.Encode(AbiEncodingStyle.Packed,
+            new AbiSignature("Transfer", AbiType.Address, AbiType.Address, AbiType.UInt256), from, to, value);
+        LogEntry? result = new(Address.Zero, data, new[] { Keccak.Zero });
```

Comment:
Are all action transfers?

Reply:
Refactored out of this scope into VM level